js/apps/admin-ui/src/user/EditUser.tsx

```diff
@@ -87,35 +87,35 @@ export default function EditUser() {
   const sessionsTab = useTab("sessions");
 
   useFetch(
-    async () => {
-      const [user, currentRealm, attackDetection] = await Promise.all([
+    async () =>
+      Promise.all([
+        adminClient.realms.findOne({ realm: realmName }),
         adminClient.users.findOne({ id: id!, userProfileMetadata: true }),
-        adminClient.realms.findOne({ realm }),
         adminClient.attackDetection.findOne({ id: id! }),
-      ]);
-
-      if (!user || !currentRealm || !attackDetection) {
+        adminClient.users.getProfileMetadata({ realm: realmName }),
```

Comment:
For edit user we don't need to query this as it's already on the use object

Reply:
Would that contain the exact same data? @pedroigor

Reply:
I'll add a commit for this now, let me know if this is wrong and I will revert it.

Reply:
@jonkoops The metadata is the same.

Reply:
Ok, I've changed this to use the data from the user instead :+1: